process/smartContract/processorV2/processV2.go

```diff
@@ -1749,32 +1835,51 @@ func (sc *scProcessor) processVMOutput(
 	return scrTxs, nil
 }
 
+type outputDataFromCall struct {
+	vmInput              *vmcommon.VMInput
+	vmOutput             *vmcommon.VMOutput
+	txHash               []byte
+	tx                   data.TransactionHandler
+	scrTxs               []data.TransactionHandler
+	acntSnd              state.UserAccountHandler
+	snapshot             int
+	createdAsyncCallback bool
+}
+
 func (sc *scProcessor) completeOutputProcessingAndCreateCallback(
-	vmInput *vmcommon.VMInput,
-	vmOutput *vmcommon.VMOutput,
-	txHash []byte,
-	tx data.TransactionHandler,
-	scrTxs []data.TransactionHandler,
-	acntSnd state.UserAccountHandler,
-	snapshot int,
-	createdAsyncCallback bool,
+	outData *outputDataFromCall,
 ) ([]data.TransactionHandler, vmcommon.ReturnCode, error) {
 
-	sc.penalizeUserIfNeeded(tx, txHash, vmInput.CallType, vmInput.GasProvided, vmOutput)
+	sc.penalizeUserIfNeeded(outData.tx, outData.txHash, outData.vmInput.CallType, outData.vmInput.GasProvided, outData.vmOutput)
 
-	scrTxs, scrForSender, scrForRelayer, err := sc.createSCRForSenderAndRelayerAndRefundGas(vmInput, vmOutput, txHash, tx, scrTxs, createdAsyncCallback)
+	scrTxs, scrForSender, scrForRelayer, err := sc.createSCRForSenderAndRelayerAndRefundGas(
```

Comment:
I think here you can send the whole outputDataFromCall DTO even if 1 or 2 parameters will not be use in destination

Reply:
done